server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/cache/CassandraDumbBlobStoreCache.java

```diff
@@ -0,0 +1,119 @@
+package org.apache.james.blob.cassandra.cache;
+
+import static com.datastax.driver.core.querybuilder.QueryBuilder.bindMarker;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.eq;
+import static org.apache.james.blob.cassandra.BlobTables.BucketBlobTable.ID;
+import static org.apache.james.blob.cassandra.BlobTables.DumbBlobCache.DATA;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+
+import javax.inject.Inject;
+
+import org.apache.james.backends.cassandra.utils.CassandraAsyncExecutor;
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.cassandra.BlobTables;
+import org.apache.james.blob.cassandra.utils.DataChunker;
+import org.reactivestreams.Publisher;
+
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.querybuilder.QueryBuilder;
+import com.google.common.base.Preconditions;
+
+import reactor.core.publisher.Mono;
+
+public class CassandraDumbBlobStoreCache implements DumbBlobStoreCache {
+
+    private final DataChunker dataChunker;
+    private final CassandraAsyncExecutor cassandraAsyncExecutor;
+    private final PreparedStatement insert;
+    private final PreparedStatement select;
+    private final PreparedStatement delete;
+    private final CassandraCacheConfiguration cacheConfiguration;
+
+    @Inject
+    public CassandraDumbBlobStoreCache(Session session, CassandraCacheConfiguration cacheConfiguration) {
+        this.dataChunker = new DataChunker();
+        this.cassandraAsyncExecutor = new CassandraAsyncExecutor(session);
+        this.insert = prepareInsert(session);
+        this.select = prepareSelect(session);
+        this.delete = prepareDelete(session);
+        this.cacheConfiguration = cacheConfiguration;
+    }
+
+    private PreparedStatement prepareDelete(Session session) {
+        return session.prepare(
+            QueryBuilder.delete().
+                from(BlobTables.BucketBlobTable.TABLE_NAME)
+                .where(eq(ID, bindMarker(ID))));
+    }
+
+    private PreparedStatement prepareSelect(Session session) {
+        return session.prepare(
+            QueryBuilder.select()
+                .from(BlobTables.BucketBlobTable.TABLE_NAME)
+                .where(eq(ID, bindMarker(ID))));
+    }
+
+    private PreparedStatement prepareInsert(Session session) {
+        return session.prepare(
+            QueryBuilder.insertInto(BlobTables.DumbBlobCache.TABLE_NAME)
+                .value(ID, bindMarker(ID))
+                .value(DATA, bindMarker(DATA)));
+    }
+
+    private boolean isAValidStream(InputStream bufferedData) throws IOException {
```

Comment:
I think you don't want to know if it's valid but if it's bigger than your threshold

Reply:
Yes, i just want to validate it bigger than threshold or not.

Reply:
Then the method name needs to reflect that ^^